### PR TITLE
Dt 137 - Add support for getting case note types

### PR DESCRIPTION
**Pull request by Mjwillis**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/casenotes/controllers/CaseNoteController.java [hunk 1]
```diff
@@ -102,4 +99,23 @@ public CaseNote amendCaseNote(
 
        return caseNoteService.getCaseNotes(filter, pageable);
     }
+
+    @GetMapping("/types")
+    @ApiOperation(value = "Retrieves a list of case note types", response = CaseNoteType.class)
```
Comment (JamieDraperUK) [inline src/main/java/uk/gov/justice/hmpps/casenotes/controllers/CaseNoteController.java]: Is responseContainer=“list” needed here?

Comment (Mjwillis) [reply src/main/java/uk/gov/justice/hmpps/casenotes/controllers/CaseNoteController.java]: 👍

File: src/main/java/uk/gov/justice/hmpps/casenotes/config/RestTemplateConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+import uk.gov.justice.hmpps.casenotes.utils.JwtAuthInterceptor;
+import uk.gov.justice.hmpps.casenotes.utils.W3cTracingInterceptor;
+
+import java.util.List;
+
+@Configuration
+public class RestTemplateConfiguration {
+
+    private final OffenderCaseNoteProperties properties;
+    private final OAuth2ClientContext oauth2ClientContext;
```
[the remark this answers is not in the document]
Comment (Mjwillis) [reply src/main/java/uk/gov/justice/hmpps/casenotes/config/RestTemplateConfiguration.java]: Deleted as don't need this at the moment

File: src/main/java/uk/gov/justice/hmpps/casenotes/controllers/CaseNoteController.java (hunk 1 above)
Comment (Mjwillis) [reply src/main/java/uk/gov/justice/hmpps/casenotes/controllers/CaseNoteController.java]: Corrected

**Pull request merged**